python/sglang/srt/models/deepseek_v2.py

```diff
@@ -673,7 +677,26 @@ def forward_absorb(
                 q_nope_val, self.w_kc, q_nope_scale, self.w_scale, torch.bfloat16
             )
         else:
-            q_nope_out = torch.bmm(q_nope.transpose(0, 1), self.w_kc)
+            if self.use_intel_amx_backend:
+                # [Note] Align shapes of bmm inputs.
+                # Shapes of inputs:
+                #   q_nope: [M, B, K]
+                #   original self.w_kc: [B, K, N]
+                #   current self.w_kc (which has been converted in PackWeightMethod): [B, N, K]
+
+                # Shapes of inputs to sgl_kernel.cpu.bmm:
+                #   out: [B, M, N]
+                #   mat1: [B, M, K]
+                #   mat2: [B, N, K]
+                B = self.w_kc.size(0)
+                N = self.w_kc.size(1)
+                M = q_nope.size(0)
+
+                q_nope_out = torch.empty([B, M, N], dtype=q_nope.dtype)
+                sgl_kernel.cpu.bmm(q_nope_out, q_nope.transpose(0, 1), self.w_kc)
+            else:
+                q_nope_out = torch.bmm(q_nope.transpose(0, 1), self.w_kc)
+
         q_input[..., : self.kv_lora_rank] = q_nope_out.transpose(0, 1)
```

Comment:
we don't have to create q_nope_out here, bmm can accept non-contiguous out tensor, so you can pass the sub tensor from q_input to `bmm`. This will remove the memory copy in `q_input[..., : self.kv_lora_rank] = q_nope_out.transpose(0, 1)`

example: https://github.com/mingfeima/sgl-cpu-tests/blob/main/test_bmm_fp8.py#L81